src/main/scala/SerialAdapter.scala

```diff
@@ -261,18 +261,17 @@ trait CanHavePeripheryTLSerial { this: BaseSubsystem =>
       beatBytes = memParams.beatBytes
     )
 
+    // Assume we are in the same domain as our client-side binding.
     val domain = LazyModule(new ClockSinkDomain(name=Some(portName)))
+    domain.clockNode := client.fixedClockNode
 
-    // TODO: currently assume manager and client buses have same clock
-    domain.clockNode := manager.fixedClockNode
     val serdesser = domain { LazyModule(new TLSerdesser(
       w = params.width,
       clientPortParams = clientPortParams,
       managerPortParams = managerPortParams
     )) }
     manager.coupleTo(s"port_named_serial_tl_mem") {
-      (serdesser.managerNode
-        := TLBuffer()
+      ((domain.crossIn(serdesser.managerNode)(ValName("TLSerialManagerCrossing")))(p(SerialTLAttachKey).slaveCrossingType)
```

Comment:
Do crossings include a buffer by default? I think we usually want to include one in this path.

Reply:
Yes, if the crossing type is "SynchronousCrossing" and not "NoCrossing".